packages/performance/docs/events.md

```diff
@@ -0,0 +1,126 @@
+# Events
+
+## Lifecycle Events
+
+### Time to First Byte
+
+Use enum `EventType.TimeToFirstByte`
+
+The time until the server sends the first part of the response.
+[Learn more](https://developers.google.com/web/tools/chrome-devtools/network-performance/understanding-resource-timing)
+
+### First Paint
+
+Use enum `EventType.TimeToFirstPaint`
+
+The time until the browser renders anything that differs from the previous page.
+[Learn more](https://developers.google.com/web/fundamentals/performance/user-centric-performance-metrics#first_paint_and_first_contentful_paint)
+
+### First Contentful Paint
+
+Use enum `EventType.TimeToFirstContentfulPaint`
+
+The time until the browser renders the first bit of content from the DOM.
+[Learn more](https://developers.google.com/web/tools/lighthouse/audits/first-contentful-paint)
+
+### DOM Content Loaded
+
+Use enum `EventType.DomContentLoaded`
+
+The time until the DOM has been entirely loaded and parsed.
+[Learn more](https://developer.mozilla.org/en-US/docs/Web/Events/DOMContentLoaded)
+
+### First Input Delay
+
+Use enum `EventType.FirstInputDelay`
+
+The time from when a user first interacts with your site to the time when the browser is able to respond to that interaction.
+[Learn more](https://developers.google.com/web/updates/2018/05/first-input-delay)
+
+### Load Event
+
+Use enum `EventType.Load`
+
+The time until the DOM and all its styles and synchronous scripts have loaded.
+[Learn more](https://developer.mozilla.org/en-US/docs/Web/Events/load)
+
+## Navigation Events
+
+### Long Task
+
+Use enum `EventType.LongTask`
+
+Any task that take 50 milliseconds or more.
+[Learn more](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceLongTaskTiming)
+
+### Script Download Event
+
+Use enum `EventType.ScriptDownload`
+
+The time spent downloading a script resource.
+
+### Style Download Event
+
+Use enum `EventType.StyleDownload`
+
+The time spent downloading a style resource.
+
+### GraphQL Event
+
+Use enum `EventType.GraphQL`
+
+The time spent resolving GraphQL queries during the navigation.
+
+This metric needs to be manually set up in Apollo Client.
+The setup can be done as a [ApolloLink](https://www.apollographql.com/docs/link/).
+
+```tsx
```

Comment:
huh...so as I typed this out. I realized we should probably just make this link into a re-usable utility.

what does people think about between moving [`operationDetailsLink`](https://github.com/Shopify/web/blob/upgrade-packages/app/utilities/graphql-client/link/operation-details.ts#L21-L80) into `@shopify/react-graphql` and expect the consumer to add the performance part themselves. 

Or make the simplify version below into `@shopify/react-graphql`.

Reply:
didn't @ismail-syed  work on that?

Reply:
I found remanent of it in 3 different PR, but I guess non of it ever made it to production.
https://github.com/Shopify/quilt/pull/1018
https://github.com/Shopify/quilt/pull/1015
https://github.com/Shopify/quilt/pull/1005

Reply:
Yeah can we maybe take this one out until we have that link available? I don't want people copy pasting this around.